test/compilable/test21591.d

```diff
@@ -0,0 +1,46 @@
+// https://issues.dlang.org/show_bug.cgi?id=21591
+
+alias F = void function();
+
+void fn1(void function(), void function());
+void fr1(F, F);
+static assert(fn1.mangleof == "_D9test215913fn1FPFZvQeZv");
+static assert(fr1.mangleof == "_D9test215913fr1FPFZvQeZv");
+
+void fn2(void function()*, void function()*);
+void fr2(F*, F*);
+static assert(fn2.mangleof == "_D9test215913fn2FPPFZvQfZv");
+static assert(fr2.mangleof == "_D9test215913fr2FPPFZvQfZv");
+
+void function() fn3(void function()**, void function()*);
+F fr3(F**, F*);
+static assert(fn3.mangleof == "_D9test215913fn3FPPPFZvQfZQh");
+static assert(fr3.mangleof == "_D9test215913fr3FPPPFZvQfZQh");
```

Comment:
This 'reference' mangle changes as well; previously, the return type wasn't backref'd (`_D9test215913fr3FPPPFZvQfZPFZv`).